### PR TITLE
Delay key creation to avoid throttling

### PR DESCRIPTION
**Pull request by heaths**
Key creation limits are much lower than key use limits.

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using Azure.Core.Testing;
+using Castle.DynamicProxy;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Keys.Tests
+{
+    // TODO: Remove this implementation when https://github.com/Azure/azure-sdk-for-net/issues/8575 is fixed.
+    internal class DelayCreateKeyInterceptor : IInterceptor
+    {
+        private const string DelayEnvironmentVariableName = "AZURE_KEYVAULT_TEST_DELAY";
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: This should be based on the run mode. Live vs Playback.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if (_mode != RecordedTestMode.Playback && _delay > 0)
+            {
+                string name = invocation.Method.Name;
+                switch (name)
+                {
+                    case nameof(KeyClient.CreateEcKeyAsync):
+                    case nameof(KeyClient.CreateKeyAsync):
+                    case nameof(KeyClient.CreateRsaKeyAsync):
+                        TestContext.WriteLine("Delaying {0} by {1}ms", name, _delay);
+                        Thread.Sleep(_delay);
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: Delaying a second for every call is sub-optimal, can we record the fact when last key creation happened and decide how long to delay based on that?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs (hunk 1 above)
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: Who would set this?

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: It allows us to override it in the pipeline as a variable, so we can adjust without commit as needed.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs (hunk 2 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: We could store/compare that as a static, sure. Feel strongly it's worth it given this doesn't actually add all that much time to live/recording tests?

Comment (pakrym) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: Nope, not strongly. If id doesn't slow things too much I'm fine.

**Review by pakrym — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs (hunk 2 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: It wouldn't be hard. Let me play around with it a bit.

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs]: Change pushed. On my laptop while WFH, most delays were still the default max (1s) but a few came under in the mid- to high 900ms. I expect we'll see much better gains on the platform where 1 to 3 methods were failing per live run (linux).

**Pull request merged**